testscripts/perf_nw_unloadload_vmxnet3.ps1

```diff
@@ -0,0 +1,211 @@
+        default {}
+    }
+}
+
+
+# Check all parameters are valid
+if (-not $rootDir) {
+    "Warn : no rootdir was specified"
+}
+else {
+    if ( (Test-Path -Path "${rootDir}") ) {
+        Set-Location $rootDir
+    }
+    else {
+        "Warn : rootdir '${rootDir}' does not exist"
+    }
+}
+
+if ($null -eq $sshKey) {
+    "FAIL: Test parameter sshKey was not specified"
+    return $False
+}
+
+if ($null -eq $ipv4) {
+    "FAIL: Test parameter ipv4 was not specified"
+    return $False
+}
+
+if ($null -eq $logdir) {
+    "FAIL: Test parameter logdir was not specified"
+    return $False
+}
+
+
+# Source tcutils.ps1
+. .\setupscripts\tcutils.ps1
+PowerCLIImport
+ConnectToVIServer $env:ENVVISIPADDR `
+    $env:ENVVISUSERNAME `
+    $env:ENVVISPASSWORD `
+    $env:ENVVISPROTOCOL
+
+
+#######################################################################################
+## Main Body
+#######################################################################################
+# Current version only install sshpass by rpm link
+$retValdhcp = $Failed
+
+$linuxOS = GetLinuxDistro $ipv4 $sshKey
+$linuxOS = $linuxOS[-1]
+$vmOut = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+$vmNameB = $vmName -replace "A","B"
+Write-Host -F Red "INFO: RevertSnap $vmNameB..."
+Write-Output "INFO: RevertSnap $vmNameB..."
+$result = RevertSnapshotVM $vmNameB $hvServer
+if ($result[-1] -ne $true)
+{
+    Write-Host -F Red "INFO: RevertSnap $vmNameB failed"
+    Write-Output "INFO: RevertSnap $vmNameB failed"
+    DisconnectWithVIServer
+    return $Aborted
+}
+# Refresh status
+$vmObjB = Get-VMHost -Name $hvServer | Get-VM -Name $vmNameB
+Write-Host -F Red "INFO: Starting $vmNameB..."
+Write-Output "INFO: Starting $vmNameB..."
+# Start Guest
+Start-VM -VM $vmObjB -Confirm:$false -RunAsync:$true -ErrorAction SilentlyContinue
+Write-Host -F Red  "DEBUG: Finish start VM_B"
+# Wait for VM_B start and gei ip address
+$ret = WaitForVMSSHReady $vmNameB $hvServer ${sshKey} 300
+if ( $ret -ne $true )
+{
+    Write-Output "Failed: Failed to start VM."
+    write-host -F Red "Failed: Failed to start VM."
+    DisconnectWithVIServer
+    return $Aborted
+}
+# Refresh status
+$vmObjB = Get-VMHost -Name $hvServer | Get-VM -Name $vmNameB
+$IPB = GetIPv4ViaPowerCLI $vmNameB $hvServer
+Write-Host -F Red "DEBUG: IP address of VM_B: ${IPB}"
+Write-Output "DEBUG: IP address of VM_B: ${IPB}"
+
+# Confirm the vmxnet3 nic and get nic name
+$Command = 'lspci | grep -i Ethernet | grep -i vmxnet3' 
+$IsNIC = bin\plink.exe -i ssh\${sshKey} root@${ipv4} $Command
+Write-Host -F Red "DEBUG: Info of NIC: $IsNIC"
+Write-Output "DEBUG: Info of NIC: $IsNIC"
+if (!$IsNIC)
+{
+    Write-Host -F Red "ERROR: NIC is not vmxnets"
+    Write-Output "ERROR: NIC is not vmxnets"
+    DisconnectWithVIServer
+    return $Aborted
+}
+
+# Ping -f VMB from VMA for one hour
+$during = 3600
+$result = bin\pscp.exe -i ssh\${sshKey} remote-scripts/ping.sh root@${IPB}:/root/
```

Comment:
Can we put this ping.sh in <file> in XML, so framework will push this script to VM